src/db.c

```diff
@@ -607,7 +607,7 @@ void selectCommand(client *c) {
         addReplyError(c,"SELECT is not allowed in cluster mode");
         return;
     }
-    if (selectDb(c,id) == C_ERR) {
+    if (id < INT_MIN || id > INT_MAX || selectDb(c,id) == C_ERR) {
```

Comment:
The nitpick in me thinks this should return the same types of errors as copy() and move(), which all return different errors. This command probably has the best error messages.

We maybe should just also move id > INT_MAX into selectDB, since it already checks for less than 0 so we can omit the id < INT_MAX check.

Reply:
If  put dbid > INT_MAX into selectDB, the selectDb method needs to be changed to selectDb(client *c, long id), will this cause a misunderstanding that it can pass long?

Reply:
I find that copy(), move() return inconsistent error messages, copy returns "invalid DB index",
 move returns shared.outofrangeerr, can I change it uniformly?
I think "shared.outofrangeerr" is returned when dbid out of range, "invalid DB index" is returned when dbid exceeds server.dbnum.

Reply:
Could add a getIntFromObjectOrReply method?

Reply:
FYI, currently, we have `getRangeLongFromObjectOrReply`

Reply:
> FYI, currently, we have `getRangeLongFromObjectOrReply`

Thanks for tips.
I just tried to write getIntFromObjectOrReply.
Since this method can be used by select(), copy(), and move(), I'm debating whether to add this method.

```c
int getIntFromObjectOrReply(client *c, robj *o, int *target, const char *msg) {
    long value;

    if (getRangeLongFromObjectOrReply(c, o, INT_MIN, INT_MAX, &value, msg) ! = C_OK) 
        return C_ERR;

    *target = value;
    return C_OK;
}
```

Reply:
@ShooterIT If  use getRangeLongFromObjectOrReply directly, the target is  long type, which may not be readable for the code.